src/GraphQL.SystemTextJson/ObjectDictionaryConverter.cs

```diff
@@ -0,0 +1,223 @@
+                    continue;
+                }
+
+                WriteProperty(writer, options, value, entry.Key);
+            }
+
+            writer.WriteEndObject();
+        }
+
+        private void WriteProperty(Utf8JsonWriter writer, JsonSerializerOptions options, object value, string key)
+        {
+            switch (value)
+            {
+                case int intValue:
+                    writer.WriteNumber(key, intValue);
+                    break;
+                case double doubleValue:
+                    writer.WriteNumber(key, doubleValue);
+                    break;
+                case decimal decimalValue:
+                    writer.WriteNumber(key, decimalValue);
+                    break;
+                case string stringValue:
+                    writer.WriteString(key, stringValue);
+                    break;
+                case bool boolValue:
+                    writer.WriteBoolean(key, boolValue);
+                    break;
+                case Dictionary<string, object> subDictionary:
+                    writer.WritePropertyName(key);
+                    WriteDictionary(writer, subDictionary, options);
+                    break;
+                case IEnumerable list:
+                    writer.WritePropertyName(key);
+                    WriteArray(writer, list, options);
+                    break;
+                default:
+                    JsonSerializer.Serialize(writer, value, value.GetType(), options);
+                    break;
+            }
+        }
+
+        private void WriteArray(Utf8JsonWriter writer, IEnumerable list, JsonSerializerOptions options)
+        {
+            writer.WriteStartArray();
+
+            foreach (var value in list)
+            {
+                WriteArrayValue(writer, value, options);
+            }
+
+            writer.WriteEndArray();
+        }
+
+        private void WriteArrayValue(Utf8JsonWriter writer, object value, JsonSerializerOptions options)
+        {
+            switch (value)
+            {
+                case int intValue:
+                    writer.WriteNumberValue(intValue);
+                    break;
+                case double doubleValue:
+                    writer.WriteNumberValue(doubleValue);
+                    break;
+                case decimal decimalValue:
+                    writer.WriteNumberValue(decimalValue);
+                    break;
+                case string stringValue:
+                    writer.WriteStringValue(stringValue);
+                    break;
+                case bool boolValue:
+                    writer.WriteBooleanValue(boolValue);
+                    break;
+                case Dictionary<string, object> subDictionary:
+                    WriteDictionary(writer, subDictionary, options);
+                    break;
+                case IEnumerable list:
+                    WriteArray(writer, list, options);
+                    break;
+            }
```

Comment:
Copy/paste from `WriteProperty`. Consider to move inside method like `WriteValue` + there is no default case :(

Reply:
Done - will push shortly.